utils/prod/update_config.py

```diff
@@ -29,48 +30,69 @@
 ]
 
 
+def get_nested_dict(data: Dict[str, Any], keys: List[str]) -> Optional[Dict[str, Any]]:
+    """Retrieve a nested dictionary given a list of keys."""
+    for key in keys:
+        if isinstance(data, dict) and key in data:
+            data = data[key]
+        else:
+            return None
+    return data if isinstance(data, dict) else None
+
+
+def update_nested_dict(data: Dict[str, Any], keys: List[str], value: Any) -> None:
+    """Update a nested dictionary given a list of keys and a value."""
+    for key in keys[:-1]:
+        data = data.setdefault(key, {})
+    data[keys[-1]] = value
+
+
 def update_embedding_model(config: Dict[str, Any]) -> None:
     """Update the embedding model configuration."""
-    if 'embedding_model' in config:
-        embedding_model = config['embedding_model']
-        if isinstance(embedding_model, dict):
-            embedding_model['type'] = 'sambastudio'
-            embedding_model['batch_size'] = 32
-            embedding_model['coe'] = False
+    embedding_model = get_nested_dict(config, ['rag', 'embedding_model']) or get_nested_dict(
+        config, ['embedding_model']
+    )
+    if embedding_model:
+        embedding_model['type'] = 'sambastudio'
+        embedding_model['batch_size'] = 32
+        embedding_model['coe'] = False
 
 
 def update_config_prod(config: Dict[str, Any]) -> None:
     """Update configuration for production mode."""
-    if 'prod_mode' in config:
-        config['prod_mode'] = True
-
+    update_nested_dict(config, ['prod_mode'], True)
```

Comment:
Here we enforce the prod mode. We should explain why.

Reply:
Enforcing prod_mode ensures that the application runs with production settings, which might include optimizations and configurations different from development mode. I'll add a comment to explain the importance of this enforcement.